### ISSUE TITLE
On high traffic the httpd container is not working

### ISSUE THREAD

**Issue by uGiFarukh** [cytopia/devilbox#208]
I have running a docker engine on the latest stable CoreOS in a baremetal. My server configuration is 8 Core @ 3.8GHz Intel Xeon 32GB RAM 10Gbps network and 240GB SSD. It should run quite a lot of traffic. But I used the devilbox LAMP stack on my docker for my high traffic website. But after sometime of start the httpd becomes unresponsive. I couldn't go to my url anymore. Why is that happening? More than half of my server resources din't even got used up. Please help me out find why the httpd seems to be crashing.

- Running latest repo.
- PHP7.0, MySQL-5.7, httpd2.4 that's all.
- Attached all the logs below.
- very normal docker-compose output:
 docker-compose down
Stopping laxmp_mysql_1 ... done
Stopping laxmp_httpd_1 ... done
Stopping laxmp_php_1   ... done
Stopping laxmp_bind_1  ... done
Removing laxmp_mysql_1 ... done
Removing laxmp_httpd_1 ... done
Removing laxmp_php_1   ... done
Removing laxmp_bind_1  ... done
Creating laxmp_bind_1  ... done
core@cfd ~/LAXMP $ docker-compose up -d httpd php mysql
Creating laxmp_php_1   ... done
Creating laxmp_bind_1  ...
Creating laxmp_httpd_1 ... done
Creating laxmp_httpd_1 ...
Creating laxmp_mysql_1 ... done

- No use of different version of database. Just a fresh devilbox install.

Please also specify the following info:

- CoreOS 1632.3
- 17.09.1
- 1.19.0

**Comment by cytopia** [comment]
@uGiFarukh is anything actually crashing or is it just unresponsive?

The general design of the devilbox is to use it for local development, so all settings are optimized for that scenario. If you want to use it in high load scenario's I guess the PHP-FPM and the Apache base configuration files need to be tweaked and adjusted to your processor and RAM.

Also to note, that logging should be minimized to warning only to reduce overhead as well PHP.ini needs to be adjusted.

If you provide any values for php and apache that are suiteable for your machine, I will make some changes so it is easier to tweak the containers.

**Comment by cytopia** [comment]
I would also suggest you stick to this branch as it will get merged very soon (it has completely redesigned PHP container): https://github.com/cytopia/devilbox/pull/203

**Comment by uGiFarukh** [comment]
Actually the docker container still keep on running. Only nothing shows up on the port 80. I tried testing port 3306 to see if it's a mysql problem but the port 3306 is healthy and working. I did a bit of tweaking on the php and apache configurations. But I think that's not enough for a production server. Can you upload some good tweaks to the php.ini, apache.conf and my.cnf files suitable for production environment? How can I minimize the logging system to warning only? I am not sure about this one.

I myself not sure what might be suitable php and apache values for a production environment. But I will look into it and let you know if I find something.

And I have head the new php container news I have read the changes on that pull request. Seems a good approach by using the most latest official php docker images at the base. I hope you are also using the official docker latest images of httpd, mysql and nginx along with php.

Suppose, I should mention that on my server, I have modified the whole devilbox repo heavy by removing the unnecessary config variables and docker yaml commands for my build. And, also I have removed the devilbox monitor API which was assigned as the default template. If you allow me I can upload the files to a different dedicated branch to this project for production use with all the optimizations and configs. Really looking forward to see the use of base official images in all the containers.

**Comment by uGiFarukh** [comment]
And maybe I can try putting some base config for the production build which can be easily scaled and load balanced throughout coreos clusters or the production environment.

**Comment by cytopia** [comment]
Feel free to push it to some branch, would be good to have it. I am also planing a `docker-compose-prod.yml` for this kind of stuff. The new PHP images already have a production image each as well as a development image.

So in your case, first step would be to verify if the latency comes from php files or only from the webserver. In times it is not responding, try to fetch a static document from it (such as css or js file) and see if this works or if not.

Another thing after having done the above finding, would be to give nginx a try and see if it behaves better.

**Comment by uGiFarukh** [comment]
Alright I will do a push after some research on the production LAMP.

Got it. will try to fetch some static file when it will become unresponsive. And will share my feedback here.

I was planning to give nginx a try but due to my limited knowledge in nginx scripting I decided to put it aside for a while. But if I got some good research outcome on the production build I will also look into nginx in production.

**Comment by uGiFarukh** [comment]
Hey @cytopia do you have any gitter or discord channel where we can chat? I really wanna push the production version today because my website is acting weird in the current build. Needed to discuss some stuffs before I do. And, also I was thinking of putting a simple FTP container with LAMP stack.

**Comment by cytopia** [comment]
gitter is listed right on the front of the readme: https://gitter.im/devilbox/Lobby?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge

**Comment by uGiFarukh** [comment]
yeah I tried an static html file in the root directory of the domain. The server cannot read this file also, But while this happens if I put the html file in devilbox template directory and try to load it. It works. Really weird.

**Comment by cytopia** [comment]
Hmm this is very strange, it shouldn't actually be able to serve anything from the devilbox template directory. Could you paste the actual paths for project dir, htdocs dir and template dir.

**Comment by uGiFarukh** [comment]
Can we discuss this over gitter? And also I will be needing some of your help for the production stack. I haven't been able to get my head around your entrypoint scripts. Some alpine production images would be very good to have in the production stack.

**Comment by cytopia** [comment]
Guess we're in different timezones and its hard to catch each other. I would suggest you provide all the required information and we work through it step by step and see that we can catch each other on gitter as well.

**Comment by uGiFarukh** [comment]
project dir: /home/core/www/web/projectdomain

htdocs dir: /home/core/LAXMP/.devilbox/www/htdocs

Template dir: /home/core/LAXMP/.devilbox/www/htdocs/

**Comment by uGiFarukh** [comment]
I will proceed with this debugging on a leisure time. Currently working on the production stack. @cytopia Can you please check my recent gitter message because I am having a problem php error with the stack using your php prod and httpd latest.

**Comment by cytopia** [comment]
@uGiFarukh while make the Devilbox also configurable PHP-FPM server settings, I hit the same problem that PHP was not responsive. It looks like it was due to the fact, that too little FPM childs started.

I have increased the default number of childs:

https://github.com/devilbox/docker-php-fpm/commit/8b49cd618474cddf2b40610285d04bf226a55366

As well documented how to change it during run-time:

http://devilbox.readthedocs.io/en/documentation/configuration-files/php-fpm-conf.html

**Comment by cytopia** [comment]
@uGiFarukh the new release is out and it should resolve the performance issues. Can u give it a try and report back here.

**Comment by uGiFarukh** [comment]
Sure I will try and let you know soon.

**Comment by cytopia** [comment]
@uGiFarukh any update on that?
